extension-manifest-v3/src/background/stats.js

```diff
@@ -16,13 +17,20 @@ import { getOffscreenImageData } from '@ghostery/ui/wheel';
 import { order } from '@ghostery/ui/categories';
 
 import Request from './utils/request.js';
-import { getMetadata } from './utils/trackerdb.js';
+import * as trackerDb from './utils/trackerdb.js';
 import Options, { observe } from '/store/options.js';
 import AutoSyncingMap from './utils/map.js';
 
-// If you bump this number, the extension will start with a
-// clean state. Normally, this should not be needed.
 const tabStats = new AutoSyncingMap({ storageKey: 'tabStats:v1' });
+const dailyStats = new AutoSyncingMap({
+  storageKey: 'dailyStats:v1',
+  softFlushIntervalInMs: 1000, // 1 second
+  hardFlushIntervalInMs: 5000, // 5 seconds
+  ttlInMs: 1000 * 60 * 60 * 24 * 365 * 5, // 5 years
+  maxEntries: 365 * 5 + 2, // ~ 5 years of data + 2 days for 29th of February (worst case scenario)
```

Comment:
If entries are added each day, you might want to add a bit of buffer here and for the cleanup rely on the TTL to expire. If maxEntries hit, it will purge all entries, which means all history would be lost. (maxEntries is kind of a panic mode to detect cases that should never happen and then forces a reset).

Reply:
I made a test of setting object with 2500 entries with the structure of the daily stats, and the average from 10 tries is about **21ms** on my machine (unfortunately slowing down the CPU in dev tools does not help - it means, that the core of the work is done in native part). 

* 2500 ~ 21ms
* 1500 ~ 13ms
* 500 ~ 8ms

I assume that on slower machines it might be longer, but still I don't expect anything more than 50ms - so it is quite fast ;)

Reply:
think of 2018 iPhone, or a mid-range Android device

Reply:
Simulated data of one day with 280 different trackers:

<img width="319" alt="Screenshot 2023-03-28 at 13 40 23" src="https://user-images.githubusercontent.com/1228153/228224873-3d8d80c9-d922-4cc5-9c00-d474c6653ca1.png">

Fairly pessimistic estimation:
5 years = 1825 * 2586 bytes = 4,7 MB